### PR TITLE
[benchmark] Add versions of prims that in a non-efficient way uses va…

### PR DESCRIPTION
**Pull request by gottesmm**
…rious forms of non-strong references.

Specifically, I add some benchmarks for weak, unowned, unsafe (unowned), and
unmanaged. The reason for the split in between unsafe (unowned) and unmanaged is
that one is testing the raw compiler features and the other is validating stdlib
performance.

### CODE REVIEW

**Comment by gottesmm** [comment]
NOTE: I realize the doctor is going to tell me some things to do around naming/etc

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
Fixed up the names and hoisted out the initial graph/map creation. Lets see if that makes a difference.

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
@swift-ci smoke benchmark

**Comment by gottesmm** [comment]
Hmm... I hoisted out all of the setup stuff for all 4 types... wonder what is special about weak that makes that not enough.

File: benchmark/single-source/PrimsNonStrongRef.swift [hunk 1]
```diff
@@ -0,0 +1,1109 @@
...
+
+  func free()
+}
+
+extension ValueBox {
+  func free() {}
+}
+
+protocol GraphNode : AnyObject {
+  associatedtype BoxType : ValueBox where BoxType.ValueType : GraphNode
```
Comment (nate-chandler) [inline benchmark/single-source/PrimsNonStrongRef.swift]: Would it make sense for the where clause to be `where BoxType.ValueType == Self`?

Comment (gottesmm) [reply benchmark/single-source/PrimsNonStrongRef.swift]: I can fix that. Thanks!

**Comment by gottesmm** [comment]
I actually want to use this just to benchmark unmanaged, unowned. Weak was just a nice add on. To fix Weak I need to change the workload/algorithm. I would rather just leave weak for another time. So I am going to do that then smoketest/merge.

**Comment by gottesmm** [comment]
(specifically, the actual input data would have to be changed, I don't want to recompute that right now)

File: benchmark/single-source/PrimsNonStrongRef.swift (hunk 1 above)
Comment (gottesmm) [reply benchmark/single-source/PrimsNonStrongRef.swift]: Done. Allowed me to simplify a bunch of other constraints actually.

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Comment by gottesmm** [comment]
@swift-ci smoke test linux platform

**Pull request merged**